Assets/LeapMotion/Scripts/Query/DirectQueryOps.cs

```diff
@@ -361,5 +361,17 @@ public partial struct QueryWrapper<QueryType, QueryOp> where QueryOp : IQueryOp<
         list.Add(obj);
       }
     }
+
+    public void FillHashSet(HashSet<QueryType> hashSet) {
+      hashSet.Clear();
+      AppendHashSet(hashSet);
+    }
+
+    public void AppendHashSet(HashSet<QueryType> hashSet) {
+      QueryType obj;
+      while (_op.TryGetNext(out obj)) {
```

Comment:
You need to make a copy of _op before you use it to allow Queries to be re-used.